sklearn/linear_model/sag.py

```diff
@@ -89,10 +88,12 @@ def sag_solver(X, y, sample_weight=None, loss='log', alpha=1.,
     sample_weight : array-like, shape (n_samples,), optional
         Weights applied to individual samples (1. for unweighted).
 
-    loss : 'log' | 'squared'
-        Loss function that will be optimized.
-        'log' is used for classification, like in LogisticRegression.
-        'squared' is used for regression, like in Ridge.
+    loss : 'log' | 'squared' | 'multinomial'
```

Comment:
Did you checked `get_auto_step_size`'s docstring?

about  `max_squared_sum`, I don't know in which case it would be useful, but we definitely move it to utils, maybe with a caller function to check the input.